app/src/main/java/com/vero/tasky/core/data/local/UserPreferencesImpl.kt

```diff
@@ -0,0 +1,42 @@
+package com.vero.tasky.core.data.local
+
+import android.content.Context
+import androidx.core.content.edit
+import androidx.security.crypto.EncryptedSharedPreferences
+import androidx.security.crypto.MasterKeys
+import com.vero.tasky.core.domain.local.UserPreferences
+import com.vero.tasky.core.domain.model.User
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+
+class UserPreferencesImpl @Inject constructor(
+    @ApplicationContext context: Context
+) : UserPreferences {
+
+    private val masterKey = MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC)
+
+    private val pref = EncryptedSharedPreferences.create(
+        "com.vero.tasky1",
+        masterKey,
+        context,
+        EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+        EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+    )
```

Comment:
I'd probably prefer injecting these preferences instead of the context. That way you have the option to reuse the same prefs instance for different types of preferences in your app. However in case of encrypted prefs it's likely you'll only need this here, so feel free to keep this